src/Converter.cpp

```diff
@@ -0,0 +1,93 @@
+#include "Converter.hpp"
+#include "frame_helper/FrameHelper.h"
+#include "base-logging/Logging.hpp"
+#include "opencv2/core/core.hpp"
+#include "opencv2/highgui/highgui.hpp"
+
+namespace log2video
+{
+
+Converter::Converter()
+{
+
+}
+
+void Converter::convert(std::string input_file, std::string stream, std::string output_file)
+{
+
+    std::cout << "Reading input file: " << input_file << std::endl;
+    pocolog_cpp::LogFile log_file_(input_file.c_str());
+
+    frame_stream_ = dynamic_cast<pocolog_cpp::InputDataStream*>(&log_file_.getStream(stream));
+    // get total number of samples
+    size_t samples = frame_stream_->getSize();
+
+    // get frames resolution
+    int width, height;
+    getResolution(frame_stream_, width, height);
+
+    //Compute FPS
+    double fps = computeFPS(frame_stream_, samples);
+
+    cv::VideoWriter output_video(output_file, CV_FOURCC('M','P','4','2'), fps,
+            cv::Size(width, height));
+
+    if(!output_video.isOpened())
+        throw std::runtime_error("The output video was not correctly opened");
+
+    std::cout << "Output video: " << output_file << std::endl;
+    std::cout << "fps: " << fps << std::endl;
+    std::cout << "Total number of frames: " << samples << std::endl;
+    std::cout << "Frames resolution: " << width << " x " << height << std::endl;
+
+    for( size_t index = 0; index < samples; ++index)
+    {
+        try
+        {
+            base::samples::frame::Frame input_frame;
+            memset(&input_frame, 0, sizeof(base::samples::frame::Frame));
```

Comment:
using memset in objects (anything that is not a primitive or a POD) is wrong in c++, potentially very dangerous and is considered a bug (undefined behavior) because you may mess with something that is used internally by the class or even the vtable if the class has virtual methods. 

Reference: http://en.cppreference.com/w/cpp/string/byte/memset

Reply:
In this particular case, you definitely break the `std::vector` which internally has pointers.